main.go

```diff
@@ -539,6 +532,21 @@ func registerWebhooks(mgr manager.Manager) {
 		))
 	}
 
+	// NOTE: AzureManagedCluster,AzureManagedControlPlane, and AzureManagedMachinePool are behind AKS feature gate flag;
+	// the webhook is going to prevent creating or updating new objects in case the feature flag is disabled
+	if err := (&infrav1exp.AzureManagedCluster{}).SetupWebhookWithManager(mgr); err != nil {
```

Comment:
AFAIK the above `hookServer.Register` were supposed to register those webhooks using custom logic with a Client (to be able to inspect other objects), it seems strange that we're doing both side by side now, is it not enough to remove line 519 so it's no longer conditional on the feature flag being enabled?

Reply:
I was copying the approach of AzureMachinepool (L507, above).

If these lines aren't needed because L519-532 is doing the necessary registration, does that suggest that we're missing the registration of the AzureManagedCluster resource type? (I only see AzureManagedMachinePool and AzureManagedControlPlane in there)

Reply:
Sorry for the delayed response here...

AzureMachinePool is different because it doesn't need the Client as input so it's not a custom webhook registration (compare `ValidateCreate(client client.Client)` vs `ValidateCreate()`)

The two that have the special registration are `AzureManagedControlPlane` and `AzureManagedMachinePool`, are those the two that are also not working?

Reply:
this is where the webhook implementation lives for those that need the Client to get other CRDs: https://github.com/kubernetes-sigs/cluster-api-provider-azure/tree/main/util/webhook

Reply:
I think I understand what you mean now. I've down-scoped this PR to just the part where we standardize to throwing webhook errors on create (so we get a meaningful feature-gate-specific error, and not a confusing "this CRD doesn't exist" error).

I think there may be edge cases for AzureManagedMachinePool, but @nojnhuh and I need to do more work digging into that.

This PR is re-ready for review.

Reply:
perfect, thanks

can you please update PR description + release note?

Reply:
done